### DISCUSSION TITLE
Can't seem to get Lap 1 data for any of the races?

### DISCUSSION THREAD

**Discussion by DilonFernando03** [br-g/openf1#103] [Q&A] [answered]
Heya folks, I was just trying to create a data visualization dashboard based on F1 driver data for my final year project but I seem to be running into issues when selecting lap 1 to visualize? All other laps work as normal but every lap 1 has a lap_duration of null? Cheers for any help.

**Comment by guid118** [comment, marked as answer]
The lap duration is null, because the sector 1 time is not recorded. Only sectors 2 and 3 are available

**Comment by denisrouleau** [comment]
The way I got lap 1 times was to get the lap 2 date_start and subtract the race start time which I got from first GREEN flag (from race control)  after the  scheduled race start (date_start) . I'm new to this API so there might be a better way.

In short
get race session date_start from session
get first GREEN flag after that time from race_control. (Flag or event may be different when starting behind safety car) This is the RACE date_start.
.
for each driver, compute  lap_1 lap_duration by taking their lap2date_start and subtracting  the RACE date_start.

**Reply by DilonFernando03** [reply]
Alrighty, thank you so much for the answer. I'll try it out!
